amida-tech/api-boilerplate
# Feature/upgrade packages

#### What's this PR do?
## Security vulnerability fixes
![Screen Shot 2019-08-14 at 12 28 47 AM](https://user-images.githubusercontent.com/8482889/62994553-83c2db00-be2a-11e9-8e5f-5d397c335d6f.png)
- Primary affect appears to have been lodash, eslint, and sequelize. The others might be sub-package dependencies (i'm not sure how the security alerts organizes itself)
- All primary affected packages have been upgraded to latest

## Upgrades
- Babel to Babel7
- Gulp to Gulp4
- Sequelize to Sequelize5
- Winston to Winston3 with all the necessary refactoring:
  - leverages Winston's logger container so multiple custom loggers can be utilized
  - logger formatter for prettiness

## Deprecation fixes
- `supertest-as-promised` is now simply `supertest`:
  - https://www.npmjs.com/package/supertest-as-promised (note the red banner deprecation message)
- `sequelize` finally removed `findById` after many versions warning of its deprecation and so I simply used `findOne` with a `where` clause instead
- `gulp` no longer supports the array with task-chaining syntax and instead implements `gulp.series` and `gulp.parallel` as improved and intuitive ways to call your tasks
  - https://codeburst.io/switching-to-gulp-4-0-271ae63530c0
  - https://fettblog.eu/gulp-4-parallel-and-series/
  - https://www.sitepoint.com/how-to-migrate-to-gulp-4/
  - This also obviates the need to use `run-sequence` and so it was removed
- `sequelize` also removed the need to define `operatorsAliases` as `false` in v5 (https://github.com/sequelize/sequelize/issues/8417#issuecomment-480547402) which was traditionally used to silence the complaints about not using string-based operators (https://github.com/sequelize/sequelize/issues/8417#issuecomment-334056048)

## Etc
- `yarn eslint --fix`ed the repo so there's a bunch of style fixes in here, too 

#### Related JIRA tickets:
- N/A

#### How should this be manually tested?
- `yarn start` and make sure it doesn't blow up
- `yarn test` and make sure everything passes

#### Any background context you want to provide?
- It was time to get this repo to current standards and best practices

#### Screenshots (if appropriate):
